front/src/common/Pathfinding/TypeAndPath.tsx

```diff
@@ -0,0 +1,179 @@
+      const constraint = opTrigrams.reduce(
+        (res, trigram) => [...res, ['=', ['trigram'], trigram]],
+        ['or'] as (string | SearchConstraintType)[]
+      );
+      const limitToMainStationConstraint = [
+        'or',
+        ['=', ['ch'], ''],
+        ['=', ['ch'], 'BV'],
+        ['=', ['ch'], '00'],
+      ];
+      const payload: PostSearchApiArg = {
+        searchPayload: {
+          object: 'operationalpoint',
+          query: ['and', constraint, ['=', ['infra_id'], infraId], limitToMainStationConstraint],
+        },
+        pageSize: 1000,
+      };
+      postSearch(payload)
+        .unwrap()
+        .then((results) => {
+          const operationalPoints = [...results] as SearchResultItemOperationalPoint[];
+          setOpList(
+            opTrigrams.map(
+              (trigram) => operationalPoints.find((op) => op.trigram === trigram) || { trigram }
+            ) as SearchResultItemOperationalPoint[]
+          );
+        });
+    }
+  }
+
+  const isInvalid = !opList.every((op) => op.name || op.trigram === '');
+
+  function launchPathFinding() {
+    if (infraId && rollingStockId && opList.length > 0) {
+      const params = {
+        infra: infraId,
+        rolling_stocks: [rollingStockId],
+        steps: opList
+          .filter((op) => op.trigram !== '')
+          .map((op) => ({
+            duration: 0,
+            waypoints: op.track_sections.map((position) => ({
+              track_section: position.track,
+              offset: position.position,
+            })),
+          })),
+      };
+      postPathfinding({ pathQuery: params })
+        .unwrap()
+        .then((itineraryCreated: Path) => {
+          zoomToFeature(bbox(itineraryCreated.geographic));
+          loadPathFinding(itineraryCreated, dispatch);
+        })
+        .catch((e) => {
+          dispatch(
+            setFailure({
+              name: e.data.name,
+              message: e.data.message,
+            })
+          );
+        });
+    }
+  }
+
+  useEffect(() => {
+    if (debouncedInputText !== '') {
+      getOpNames();
+    } else {
+      setOpList([]);
+    }
+  }, [debouncedInputText]);
+
+  return (
+    <div
+      className="type-and-path"
+      style={{ minWidth: `${monospaceOneCharREMWidth * inputText.length + 5.5}rem` }} // To grow input field & whole div along text size
+    >
+      <div className="help">{opList.length === 0 && t('inputOPTrigrams')}</div>
+      <OpTooltips opList={opList} />
+      <div className="d-flex align-items-center">
+        <div
+          className={cx('form-control-container', 'flex-grow-1', 'mr-2', {
+            'is-invalid': isInvalid,
+          })}
+        >
+          <input
+            className="form-control form-control-sm text-zone"
+            type="text"
+            value={inputText}
+            onChange={(e) => handleInput(e.target.value)}
+            placeholder={t('inputOPTrigramsExample')}
+            autoFocus
```

Comment:
Yes I know, but this input needs some special classes to ensure same width of each letter, so I couldn't use the normal <InputSNCF />

Reply:
More complicated, not only the input has to have customClasses, the up container too (to grow along content), it would include too much changes to generalistic component for a specific case...

Reply:
Ok agreed